.github/workflows/ci.yml

```diff
@@ -52,7 +52,7 @@ jobs:
       - name: Build and test 🏗️🧪
         run: |
           mkdir -p build
-          cd build && cmake -DCREATE_TEST_TARGETS=OFF ..
+          cd build && cmake -DCREATE_TEST_TARGETS=OFF -DMINIMAL_BUILD=true ..
```

Comment:
We only really support MINIMAL_BUILD for macos and win.

Reply:
Even if macos builds fine with a full build, it will throw an exception as soon as a binary uses it!

Reply:
If we only support `MINIMAL_BUILD` on macos and win then it might be worth it to reflect this in the project by setting `MINIMAL_BUILD ON` somewhere in the CMakeLists.txt, wouldn't it?

Reply:
Yeah that could be a good idea; i'd leave it out for another PR though!
Basically, it is a runtime impediment; but i'd first like to know why we are not enforcing minimal build for windows and macos before making any change!

Reply:
Makes sense, thanks!